File: docs/src/testers/functions/management/version_history.py
# @sniptest filename=version_history.py
# @sniptest show=1-10
from notte_sdk import NotteClient

client = NotteClient()

functions = client.functions.list()

for func in functions.items:
    print(f"Function: {func.name}")
    print(f"Versions: {', '.join(func.versions)}")
    print(f"Latest: {func.latest_version}")

import os  # noqa: E402 -- hidden live assertions follow the displayed example

matched = [func for func in functions.items if func.function_id == os.environ["NOTTE_FUNCTION_ID"]]
assert len(matched) == 1
assert matched[0].latest_version in matched[0].versions
results = [bool(matched[0].versions), matched[0].latest_version in matched[0].versions]

File: docs/src/testers/functions/management/version_history.ts
// @sniptest filename=version_history.ts
// @sniptest show=1-11
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

const functions = await client.functions.list();

for (const fn of functions) {
  console.log(`Function: ${fn.name}`);
  console.log(`Versions: ${fn.versions.join(', ')}`);
  console.log(`Latest: ${fn.latest_version}`);
}

const matched = functions.filter(fn => fn.function_id === process.env.NOTTE_FUNCTION_ID);
if (matched.length !== 1) throw new Error('Owned function missing from history');
const results = [matched[0].versions.length > 0, matched[0].versions.includes(matched[0].latest_version)];
export { results };
